temporalcli/commands.workflow_test.go

```diff
@@ -446,3 +452,195 @@ func (s *SharedServerSuite) testQueryWorkflow(json bool) {
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_ToLastWorkflowTask() {
+	var wfExecutions, activityExecutions int
+	s.Worker.OnDevActivity(func(ctx context.Context, a any) (any, error) {
+		activityExecutions++
+		return nil, nil
+	})
+	s.Worker.OnDevWorkflow(func(ctx workflow.Context, a any) (any, error) {
+		workflow.ExecuteActivity(ctx, DevActivity, 1).Get(ctx, nil)
+		wfExecutions++
+		return nil, nil
+	})
+
+	// Start the workflow
+	searchAttr := "keyword-" + uuid.NewString()
+	run, err := s.Client.ExecuteWorkflow(
+		s.Context,
+		client.StartWorkflowOptions{
+			TaskQueue:        s.Worker.Options.TaskQueue,
+			SearchAttributes: map[string]any{"CustomKeywordField": searchAttr},
+		},
+		DevWorkflow,
+		"ignored",
+	)
+	s.NoError(err)
+	var junk any
+	s.NoError(run.Get(s.Context, &junk))
+	s.Equal(1, wfExecutions)
+
+	// Reset to the final workflow task
+	res := s.Execute(
+		"workflow", "reset",
+		"--address", s.Address(),
+		"-w", run.GetID(),
+		"-t", "LastWorkflowTask",
+		"--reason", "test-reset-LastWorkflowTask",
+	)
+	require.NoError(s.T(), res.Err)
+	s.awaitNextWorkflow(searchAttr)
+	s.Equal(2, wfExecutions, "Should re-executed the workflow")
+	s.Equal(1, activityExecutions, "Should not have re-executed the activity")
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_ToEventID() {
+	// We execute two activities and will resume just before the second one. We use the same activity for both
+	// but a unique input so we can check which fake activity is executed
+	var oneExecutions, twoExecutions int
+	s.Worker.OnDevActivity(func(ctx context.Context, a any) (any, error) {
+		n, ok := a.(float64)
+		if !ok {
+			return nil, fmt.Errorf("expected int, not %T (%v)", a, a)
+		}
+		switch n {
+		case 1:
+			oneExecutions++
+		case 2:
+			twoExecutions++
+		default:
+			return 0, errors.New("you've broken the test!")
+		}
+		return n, nil
+	})
+
+	s.Worker.OnDevWorkflow(func(ctx workflow.Context, a any) (any, error) {
+		var res any
+		if err := workflow.ExecuteActivity(ctx, DevActivity, 1).Get(ctx, &res); err != nil {
+			return res, err
+		}
+		err := workflow.ExecuteActivity(ctx, DevActivity, 2).Get(ctx, &res)
+		return res, err
+	})
+
+	// Start the workflow
+	searchAttr := "keyword-" + uuid.NewString()
+	run, err := s.Client.ExecuteWorkflow(
+		s.Context,
+		client.StartWorkflowOptions{
+			TaskQueue:        s.Worker.Options.TaskQueue,
+			SearchAttributes: map[string]any{"CustomKeywordField": searchAttr},
+		},
+		DevWorkflow,
+		"ignored",
+	)
+	require.NoError(s.T(), err)
+	var ignored any
+	s.NoError(run.Get(s.Context, &ignored))
+	s.Equal(1, oneExecutions)
+	s.Equal(1, twoExecutions)
+
+	// We want to reset to the last WFTCompleted event before the second activity, so we
+	// need to search the history for it. I could just pick the event ID, but I don't want
+	// this test to break if new event types are added in the future
+	wfsvc := workflowservice.NewWorkflowServiceClient(s.GRPCConn)
```

Comment:
Can you use the regular client already on the suite?

Reply:
Yep. I forgot about this when I made the change to the actual logic